src/terra-dev-site/app/demo/EventEmitter.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react';
+import EventEmitter from 'terra-application/lib/utils/event-emitter';
+
+const EventEmitterExample = () => {
+  const [counter, setCounter] = useState(0);
+  const [eventEmitter, setEventEmitter] = useState(() => new EventEmitter()); // eslint-disable-line no-unused-vars
```

Comment:
just curious, does react or eslint complain if we don't create the setEventEmitter variable?

Reply:
I couldn't find any online reference or example for not having the setter. I tested it by removing it and it continues to work as expected. Neither react nor eslint complained. It appears that not including the setter just means that when useState returns the getter/setter pair, this would just ignore the setter that it returns.

Reply:
sweet. i can't think of any edge cases where we'd want to swap out the eventEmitter instance

Reply:
on second thought, b/c we remove the setter, we can also pass an empty array to useEffect to so we only the setup once
```
}, []);
```

Reply:
You could use `useRef` here instead. But thinking about that led me to wonder why we export a class at all. Can we export the singleton instance and call it good? Consumers wouldn't have to worry about this at all, just use the import directly. Or am I missing something?

Reply:
@tbiethman's suggestion makes sense to me. (Now that i asked him about it 😄 )

Reply:
Though i think we should hold off on the custom hooks until we're sure about the pattern.

Reply:
Oh yes, just importing the singleton instance makes more sense. It is also easier to use for consumers. https://github.com/cerner/terra-application/pull/72/commits/4c35de80e0b282a1808de60894e5b33e086faccc